### ISSUE TITLE
Addition of custom field dict to SigmaRule

### ISSUE THREAD

**Issue by Res260** [SigmaHQ/pySigma#65]
The [Sigma Specification](https://github.com/SigmaHQ/sigma/wiki/Specification) mentions it supports "arbitrary custom fields": 
![image](https://user-images.githubusercontent.com/14599855/193434709-07fc33f0-d242-42df-9b97-2cc817968b6f.png)

So I tried adding a `stackby` field to describe the field on which to stack multiple alert of the same detection. This is a required step for creating a QRadar detection rule.

However, when using this rule:
```yaml
title: aaaBitsadmin Download
id: fad282fb-0499-4d3f-9670-55cfc950e2dc
status: experimental
description: Detects usage of bitsadmin downloading a file
references:
  - https://blog.netspi.com/15-ways-to-download-a-file/#bitsadmin
  - https://isc.sans.edu/diary/22264
tags:
  - type.indicator
  - attack.defense_evasion
  - attack.persistence
  - attack.t1197
  - attack.s0190
author: Michael Haag
logsource:
  category: process_creation
  product: Microsoft Windows Security Event Log
detection:
  selection1:
    QID: Vulnerability Discovered on Local Host
    #Filename|contains: '\bitsadmin.exe'
  selection:
    Hostname: 'bitsadmin.exe'
  condition: (selection or not selection1)
fields:
  - CommandLine
  - ParentCommandLine
falsepositives:
  - Some legitimate apps use this, but limited.
level: medium
stackby: Hostname
```

The resulting SigmaRule object does not contain any trace of `stackby`:
![image](https://user-images.githubusercontent.com/14599855/193434858-61078e19-ca56-4faa-8b44-a594968043fa.png)

I feel like there should be a property like `custom_properties` for this. What do you think?

**Comment by thomaspatzke** [comment]
Yes, that was something that was on the list but then I forgot about it. I will add this.

By the way, there's already a [QRadar backend](https://github.com/nNipsx-Sec/pySigma-backend-qradar).

**Comment by Res260** [comment]
Thanks! If you tell me where to look I might have time to try and fix it :)

> By the way, there's already a [QRadar backend](https://github.com/nNipsx-Sec/pySigma-backend-qradar).

I am aware it exists and I take inspiration from it. However, the current QRadar backend simply creates a text-based query and bundles it in a detection, which is a very limiting way to create detection logics in QRadar, as well as introducing possible performance issues. I'm building something to take advantage of the complex rule correlation engine of QRadar to address these issues and create a solid "detection as code" pipeline :)

**Comment by thomaspatzke** [comment]
Just implemented it with the `custom_attributes` attribute. I preferred the term *attribute* here to avoid confusion with the word *field* that refers to log fields in context of Sigma rules. Released in pySigma 0.8.8.
